packages/astro/src/core/build/plugins/util.ts

```diff
@@ -75,8 +75,8 @@ export function shouldInlineAsset(
 	assetPath: string,
 	assetsInlineLimit: NonNullable<BuildOptions['assetsInlineLimit']>
 ) {
-	if (typeof assetsInlineLimit === 'number') {
-		return Buffer.byteLength(assetContent) < assetsInlineLimit;
+	if (typeof assetsInlineLimit === 'number' || typeof assetsInlineLimit === 'string') {
+		return Buffer.byteLength(assetContent) < Number(assetsInlineLimit);
```

Comment:
This operation can return a `NaN`. Should we check for that too, and use the default if that's the case?

Reply:
Good catch. I've refactored this function a bit to better handle `NaN` and simplified the logic a bit to match Vite's.